managed/services/inventory/inventory_metrics.go

```diff
@@ -0,0 +1,245 @@
+			return pointer.GetString(agent.RunsOnNodeID)
+		}
+	}
+	return ""
+}
+
+func (i *InventoryMetrics) GetAgentMetrics(ctx context.Context) (metrics []Metric, err error) {
+	metrics = []Metric{}
+
+	err = i.db.InTransactionContext(ctx, nil, func(tx *reform.TX) error {
+		dbAgents, dbAgentsError := models.FindAgents(tx.Querier, models.AgentFilters{})
+
+		if dbAgentsError != nil {
+			return dbAgentsError
+		}
+
+		for _, agent := range dbAgents {
+			runsOnNodeID := ""
+			disabled := "0"
+			metricValue := float64(0)
+
+			pmmAgentID := pointer.GetString(agent.PMMAgentID)
+
+			if agent.Disabled {
+				disabled = "1"
+			}
+
+			if agent.AgentType == models.PMMAgentType {
+				if i.registry.IsConnected(agent.AgentID) {
+					metricValue = 1
+				} else {
+					metricValue = 0
+				}
+				runsOnNodeID = pointer.GetString(agent.RunsOnNodeID)
+			} else {
+				metricValue = float64(inventorypb.AgentStatus_value[agent.Status])
+				runsOnNodeID = GetRunsOnNodeIDByPMMAgentID(dbAgents, pmmAgentID)
+			}
+
+			agentMetricLabels := []string{
+				agent.AgentID,
+				string(agent.AgentType),
+				pointer.GetString(agent.ServiceID),
+				runsOnNodeID,
+				pmmAgentID,
+				disabled,
+				pointer.GetString(agent.Version),
+			}
+
+			metrics = append(metrics, Metric{labels: agentMetricLabels, value: metricValue})
+		}
+		return nil
+	})
+
+	return metrics, err
+}
+
+func (i *InventoryMetrics) GetNodeMetrics(ctx context.Context) (metrics []Metric, err error) {
+	metrics = []Metric{}
+
+	err = i.db.InTransactionContext(ctx, nil, func(tx *reform.TX) error {
+		dbNodes, dbNodesError := models.FindNodes(tx.Querier, models.NodeFilters{})
+
+		if dbNodesError != nil {
+			return dbNodesError
+		}
+
+		for _, node := range dbNodes {
+			nodeMetricLabels := []string{
+				node.NodeID,
+				string(node.NodeType),
+				node.NodeName,
+				pointer.GetString(node.ContainerName),
+			}
+
+			metrics = append(metrics, Metric{labels: nodeMetricLabels, value: serviceEnabled})
+		}
+
+		return nil
+	})
+
+	return metrics, err
+}
+
+func (i *InventoryMetrics) GetServiceMetrics(ctx context.Context) (metrics []Metric, err error) {
+	metrics = []Metric{}
+
+	err = i.db.InTransactionContext(ctx, nil, func(tx *reform.TX) error {
+		dbServices, dbServicesError := models.FindServices(tx.Querier, models.ServiceFilters{})
+
+		if dbServicesError != nil {
+			return dbServicesError
+		}
+
+		for _, service := range dbServices {
+			serviceMetricLabels := []string{
+				service.ServiceID,
+				string(service.ServiceType),
+				service.NodeID,
+			}
+
+			metrics = append(metrics, Metric{labels: serviceMetricLabels, value: serviceEnabled})
+		}
+
+		return nil
+	})
+
+	return metrics, err
+}
+
+func (i *InventoryMetricsCollector) Describe(ch chan<- *prom.Desc) {
+	prom.DescribeByCollect(i, ch)
+}
+
+func (i *InventoryMetricsCollector) Collect(ch chan<- prom.Metric) {
+	ctx, cancelCtx := context.WithTimeout(context.Background(), requestTimeout)
+	defer cancelCtx()
```

Comment:
One context with one timeout for all requests, and these requests are sequential. It may cause problems if for example first request was too slow. Maybe it makes sense to start collection for every type in separate goroutine?